premake/generator.lua

```diff
@@ -309,6 +319,13 @@ function CreateProject(config)
 			configurations({"ReleaseWithSymbols", "Release", "Debug"})
 		end
 
+		if pch_enabled then
+			pchheader(config.pch_header)
```

Comment:
Why is the header path verification different to the source file?

Reply:
They just have different input formats.

The input for `pchheader` is treated as an exact match for the #include statement in VS, so it needs to be exactly what you'll supply to #include. It can be relative to the source directory or relative to an include directory,For non-VS, premake will attempt to locate the file automatically by itself to make it appropriate for other environments.

`pchsource`, which applies for VS only, has to be given a path similar to calling `files`; relative to the working directory, which in this case means prepending our source directory to the input. Doing it this way means that `CreateProject`'s input can be relative to the source directory. That's the most intuitive way, in my opinion